python/lsst/rubintv/production/uploaders.py

```diff
@@ -493,6 +495,62 @@ def uploadPerSeqNumPlot(
 
         return uploadAs
 
+    def checkAccess(self, tempFilePrefix: str | None = 'connection_test') -> bool:
+        """Checks the read, write, and delete access to the S3 bucket.
+
+        This method uploads a test file to the S3 bucket, downloads it,
+        compares its content to the expected content, and then deletes it. If
+        any of these operations fail, it logs an error and returns ``False``.
+        If all operations succeed, it logs a success message and returns
+        ``True``.
+
+        Parameters
+        ----------
+        tempFilePrefix : `str`, optional
+            The prefix for the temporary file that will be uploaded to the S3
+            bucket, by default 'connection_test'.
+
+        Returns
+        -------
+        checkPass : `bool`
+            ``True`` if all operations succeed, ``False`` otherwise.
+        """
+        testContent = b"Connection Test"
+        dateStr = datetime.now().strftime('%Y%m%d_%H%M%S')
+        fileName = f'test/{tempFilePrefix}_{dateStr}_file.txt'
+        with tempfile.NamedTemporaryFile() as testFile:
+            testFile.write(testContent)
+            testFile.flush()
+            try:
+                self._s3Bucket.upload_file(testFile.name, fileName)
+            except (BotoCoreError, ClientError) as e:
+                self._log.exception(f"S3Uploader Write Access check failed: {e}")
+                return False
+
+        _, fixedFilePath = tempfile.mkstemp()
```

Comment:
If you change this to using a `TemporaryFile` or `NamedTemporaryFile` [context manager](https://docs.python.org/3/library/tempfile.html#module-tempfile) you can simplify it quite a bit, including removing the whole `finally` block.